### ISSUE TITLE
Error while updating to the latest version.

### ISSUE THREAD

**Issue by AdrianSkar** [canton7/SyncTrayzor#191]
Hi, I get this error when I tried to update to the latest version. Same warning when restarting the program. 
Device: Win10 x64 
This is the info it returns + the log (attached):

```
>Version: 1.1.0.227; Variant: Installed; Arch: Amd64
Path: C:\Program Files\SyncTrayzor\SyncTrayzor.exe
System.IO.FileNotFoundException: El sistema no puede encontrar el archivo especificado.

   >en Pri.LongPath.Directory.EnumerateNormalizedFileSystemEntries(Boolean includeDirectories, Boolean includeFiles, SearchOption option, String normalizedPath, String normalizedSearchPattern)
   en SyncTrayzor.Services.FilesystemProvider.EnumerateFiles(String path, String searchPattern, SearchOption searchOption) en C:\projects\synctrayzor\src\SyncTrayzor\Services\FilesystemProvider.cs:línea 83
   en SyncTrayzor.Services.Conflicts.ConflictFileManager.FindConflictsImpl(String basePath, SlimObservable`1 subject, CancellationToken cancellationToken) en C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileManager.cs:línea 137
   en SyncTrayzor.Services.Conflicts.ConflictFileManager.<>c__DisplayClass7_0.<FindConflicts>b__0() en C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileManager.cs:línea 110
--- Fin del seguimiento de la pila de la ubicación anterior donde se produjo la excepción ---
   en System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   en System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   en SyncTrayzor.Utils.SlimObservableExtensions.<SubscribeAsync>d__1`1.MoveNext() en C:\projects\synctrayzor\src\SyncTrayzor\Utils\SlimObservable.cs:línea 139
--- Fin del seguimiento de la pila de la ubicación anterior donde se produjo la excepción ---
   en System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   en System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   en SyncTrayzor.Services.Conflicts.ConflictFileWatcher.<ScanFoldersAsync>d__31.MoveNext() en C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileWatcher.cs:línea 204
--- Fin del seguimiento de la pila de la ubicación anterior donde se produjo la excepción ---
   en System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   en System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   en SyncTrayzor.Services.Conflicts.ConflictFileWatcher.<Reset>d__26.MoveNext() en C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileWatcher.cs:línea 97
--- Fin del seguimiento de la pila de la ubicación anterior donde se produjo la excepción ---
   en System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c__DisplayClass2.<ThrowAsync>b__3(Object state)
   en System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   en System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

[SyncTrayzor.zip](https://github.com/canton7/SyncTrayzor/files/85628/SyncTrayzor.zip)

**Comment by Schroedingers-Cat** [comment]
Me too.

**Comment by canton7** [comment]
Looks like the folder `VickyPC` doesn't exist. Is that true?

**Comment by AdrianSkar** [comment]
It is true, it belongs to other devices (i.e. not synchronized with this device). I used to get some "Unexpected ID folder sent" or similar messages. I though it would be caused by all devices being set to "Introductory" and then corrected those settings, but long before this update.

**Comment by canton7** [comment]
Hmm, weird, Syncthing manages to fetch the ignores for it, which implies that it exists on-disk. I wonder why I can't access it then...

**Comment by canton7** [comment]
What exactly does `C:\Users\Adrian\Desktop\Sync\PcVicky` contain?

**Comment by AdrianSkar** [comment]
<4GB on half music and half .zip files and/or regular documents

**Comment by canton7** [comment]
Oh, so it exists, contains files, everything's file, etc? Was it in use when SyncTrayzor was started?

**Comment by AdrianSkar** [comment]
I'm sorry, taht is the folder that belongs to the VickyPC's device and it was indeed synced with this one.
It is a linux pc that misses it and gived that message (Id folder sent).

**Comment by canton7** [comment]
I'm confused: which is a linux PC? Does `C:\Users\Adrian\Desktop\Sync\PcVicky` exist on the computer running SyncTrayzor, or not?

**Comment by Schroedingers-Cat** [comment]
AdrianSkar, is that folder "unshared"?

**Comment by AdrianSkar** [comment]
It exists, has files and it was a valid folder for this device (win64 using SyncTrayzor). On the other hand, a Linux device (SyncThing GTK) returned those messages but has no PcVicky folder.

**Comment by canton7** [comment]
Returned which messages?

**Comment by AdrianSkar** [comment]
If I would like to add that folder to the Linux PC(because all devices were set as introductory at first).
But that caused no error.

**Comment by canton7** [comment]
OK, that's completely irrelevant to the issue we're trying to debug :smile: 

My problem is that SyncTrayzor can't enumerate the files and folders in `C:\Users\Adrian\Desktop\Sync\PcVicky`, for some reason.

1. Can you access them manually, in Explorer?
2. Is another program using them?

**Comment by canton7** [comment]
@Schroedingers-Cat is your error exactly the same, as in the same stack trace with the same line numbers?

**Comment by AdrianSkar** [comment]
Ok, xD.

I can access them. No program is using them as far as i can tell. I can restart the system If you like.

**Comment by Schroedingers-Cat** [comment]
canton7, I send you a mail with the log files. First I think this is a regression because when I go to the version before 1.1.0 it works. 
Second I'd like to change the language of the log files to english (currently it's german), however, I've been unable to find a configuration option in the SyncTrayzor/config.xml file.

**Comment by canton7** [comment]
Thanks - I'll check my email.

The language in the log files is unfortunately driven by whatever .NET language packs you have installed - I can't control that.

**Comment by canton7** [comment]
@Schroedingers-Cat right yours is a similar problem, but caused by an `UnauthorizedAccessException`, implying that I don't have access to look at one of your folders (I'm not sure which at this stage).

I've got a workaround in place: give me 10 minutes to get it built, then I'd be grateful if you could try it :smile:

**Comment by canton7** [comment]
In about 7 minutes I'll have an intermediate build which should fix the issue here: https://ci.appveyor.com/project/canton7/synctrayzor/build/1.1.0.229/artifacts . Would you mind testing it? Thanks

**Comment by canton7** [comment]
Righty, that's built. Mind trying it and letting me know if it still crashes? Even if it doesn't crash I'd still be interested in the log (lines containing `Failed to enumerate directories in folder` or `Failed to enumerate files in folder`) to try and figure out what the underlying cause of this is.

**Comment by Schroedingers-Cat** [comment]
I'm syncing the folder %USERPROFILE/Documents, which on Win7/10 contains hidden, non accessible (system-like protected) folders called "Pictures", "Music" and "Videos". The lines you wanted me to look for only mention these folders. Basically, everybody syncing the document folder from Vista/7 should be affected by this.

**Comment by AdrianSkar** [comment]
It seems to be working now; just a message on a specific file conflict ("choose one"). 
I've attached the new log. Thank you!!!

[SyncTrayzor.zip](https://github.com/canton7/SyncTrayzor/files/85773/SyncTrayzor.zip)

**Comment by canton7** [comment]
@Schroedingers-Cat  Aah, that makes sense. 

@AdrianSkar cool! It looks like it's having trouble with `C:\Users\Adrian\Desktop\Sync\PcVicky\00.REVISAR\windows xp\Vic`

**Comment by AdrianSkar** [comment]
Ok, I'll take a look at it.

**Comment by canton7** [comment]
@AdrianSkar everything's still working fine - it's just ignoring it. Don't worry too much - it's probably got some strange permissions on it (which may mean that Syncthing can't back it up either?)

**Comment by AdrianSkar** [comment]
Explorer can't open it either. I'll check the source device files and its Syncthing settings. Tyvm!

**Comment by canton7** [comment]
(Fixed in 1.1.1)
